### PR TITLE
polybar: allow service to read from systemd PATH variable

### PR DESCRIPTION
**Pull request by saltnpepper97**
### Description

Removed hard coded **PATH** variable in order to read from systemd directly. 
This is necessary as it will allow user created scripts to be able to run when using polybar.service.
Currently there is a hard coded variable for **PATH** that does not point to all the programs on the system, causing programs called in scripts to not be found.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

**Comment by rycee** [comment]
Please fix the commit message to fit [the guidelines](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style).

Also, please add a description as to why this change is necessary.

File: modules/services/polybar.nix [hunk 1]
```diff
@@ -211,7 +211,7 @@ in {
 
       Service = {
         Type = "forking";
-        Environment = "PATH=${cfg.package}/bin:/run/wrappers/bin";
+        Environment = "";
```
Comment (rycee) [inline modules/services/polybar.nix:214]: If the `PATH` is no longer necessary, then why keep `Environment = ""`?

Comment (saltnpepper97) [reply modules/services/polybar.nix:214]: Sure I can remove it altogether. Would I have to open a new pull request??

**Comment by teto** [comment]
> Removed hard coded PATH variable in order to read from systemd directly.

Is there a default for systemd ? what is it ? (honest question, it seemed to me at least on nixos that it was quite empty)

**Comment by saltnpepper97** [comment]
> > Removed hard coded PATH variable in order to read from systemd directly.
> 
> Is there a default for systemd ? what is it ? (honest question, it seemed to me at least on nixos that it was quite empty)
I understand that however I feel as though if the PATH variable is not hard coded people can import whatever paths are needed into systemd themselves
I understand your concern however the paths that are hard coded into the module now cannot allow for a lot of user created scripts to run with polybar when running from the systemd service file. This is due to a lot of the programs being under /nix/store/ or /run/current-system/sw/bin for me at least.

**Comment by h7x4** [comment]
Would it be better to do this the opposite way?

Declare an `extraPackages` option and then either:

1. Add them to the systemd-units `PATH`

2. Create a wrapper with a `PATH`, as in the `nnn` module:

https://github.com/nix-community/home-manager/blob/4fa1ba72a39da509399cce43995e4c1e1960ce9c/modules/programs/nnn.nix#L111-L122

**Pull request closed without merging**